packages/wrangler/src/dev/start-server.ts

```diff
@@ -325,14 +325,11 @@ export async function startLocalServer({
 		if (!bundle || !format) return;
 
 		// port for the worker
-
-		if (port !== 0) {
-			await waitForPortToBeAvailable(port, {
-				retryPeriod: 200,
-				timeout: 2000,
-				abortSignal: abortController.signal,
-			});
-		}
+		await waitForPortToBeAvailable(initialPort, {
```

Comment:
if initialPort is 0, we don't have to wait for the port available though - node should just give us an available one?